File: 028.Implement strStr().py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        
        if haystack == None or needle == None or len(haystack) < len(needle):
            return -1
        if len(needle) == 0 or needle == "":
            return 0

        for i in range(0, len(haystack)):
            if len(needle) > len(haystack[i:]):
                return -1
            
            for j in range(0, len(needle)):
                if haystack[i+j] != needle[j]:
                    break
                
                if j == len(needle) - 1:
                    return i
                
        return -1

File: 028.Implement strStr().java
/*
    Implement strStr().

Returns the index of the first occurrence of needle in haystack, or -1 if needle is not part of haystack.
     */

    public int strStr(String haystack, String needle) {
        if (needle.isEmpty())
            return 0;
        for (int i=0; i<haystack.length(); i++) {
            int index = i;
            for (int j=0; j<needle.length(); j++) {
                if (haystack.length() - i < needle.length())
                    return -1;
                if (j == needle.length()-1 && haystack.charAt(index) == needle.charAt(j))
                    return i;
                if (haystack.charAt(index) != needle.charAt(j))
                    break;
                index++;
            }
        }

        return -1;
    }
